Packs/QuestKace/Integrations/QuestKace/QuestKace.py

```diff
@@ -0,0 +1,785 @@
+        return cookie
+
+    def token_request(self, url: str, headers: Optional[dict] = None, data: Optional[str] = None) -> dict:
+        """login request for initiating a connection with the product.
+        Args:
+            url: full url that the request will be sent to.
+            headers: headers of the request.
+            data: data of the request which includes username and password.
+        Returns:
+            Dictionary of the response from the product.
+        """
+        try:
+            response = requests.request("POST", url, headers=headers, data=data, verify=self._verify)
+        except requests.exceptions.SSLError:
+            err_msg = 'SSL Certificate Verification Failed - try selecting \'Trust any certificate\' checkbox in' \
+                      ' the integration configuration.'
+            raise DemistoException(err_msg)
+        except requests.exceptions.ConnectionError:
+            raise DemistoException("Invalid url , Failed to establish a connection")
+        if response.status_code == 401:
+            raise DemistoException("Error Code 401 - Invalid user or password")
+        return response.__dict__
+
+    def machines_list_request(self, filter_fields: Optional[str] = None):
+        """List of machines.
+           Returns:
+               Response from API.
+        """
+        headers = {
+            'Accept': 'application/json',
+            'x-dell-csrf-token': self._token,
+            'x-dell-api-version': '5',
+            'Cookie': self._cookie
+        }
+        if filter_fields:
+            return self._http_request("GET", url_suffix=f"/inventory/machines?filtering={filter_fields}",
+                                      headers=headers)
+        else:
+            return self._http_request("GET", url_suffix=f"/inventory/machines", headers=headers)
+
+    def assets_list_request(self, filter_fields: Optional[str] = None) -> dict:
+        """List of assets.
+           Returns:
+               Response from API.
+        """
+        headers = {
+            'Accept': 'application/json',
+            'x-dell-csrf-token': self._token,
+            'x-dell-api-version': '5',
+            'Cookie': self._cookie
+        }
+        if filter_fields:
+            return self._http_request("GET", url_suffix=f"/asset/assets?filtering={filter_fields}", headers=headers)
+        else:
+            return self._http_request("GET", url_suffix="/asset/assets", headers=headers)
+
+    def queues_list_request(self, filter_fields: Optional[str] = None) -> dict:
+        """List of queues.
+           Returns:
+               Response from API.
+        """
+        headers = {
+            'Accept': 'application/json',
+            'x-dell-csrf-token': self._token,
+            'x-dell-api-version': '5',
+            'Cookie': self._cookie
+        }
+        if filter_fields:
+            return self._http_request("GET", url_suffix=f"/service_desk/queues?shaping=fields all"
```

Comment:
let's build the `url_suffix` accordingly to the `filter_fields`
and use only 1 return in the function